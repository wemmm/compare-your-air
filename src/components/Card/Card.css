.card {
  background-color: white;
  border-radius: 8px;
  box-sizing: border-box;
  margin-top: 10px;
  padding: 0 30px 30px;
  text-align: left;
  width: 400px;
}

.card--city {
  margin: 8px 0;
}

.card--close-button-container {
  display: flex;
  justify-content: flex-end;
  left: 20px;
  position: relative;
  top: 10px;
  -webkit-justify-content: flex-end;
}

.card--close-button {
  border: none;
  background-color: transparent;
  cursor: pointer;
  padding: 0;
}

.card--close-button:hover svg path {
  fill: #4475af;
}

.card--updated-time {
  font-size: small;
  font-weight: 600;
  line-height: 0.2;
}

@media (max-width: 810px) {
  .card:first-of-type {
    margin-top: 20px;
  }
}
