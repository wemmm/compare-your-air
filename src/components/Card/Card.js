import { formatDistanceToNow } from "date-fns";

const Card = ({ cityData }) => {
  return (
    <div className="card">
      <p>
        UPDATED{" "}
        {formatDistanceToNow(
          new Date(cityData.measurements[0].lastUpdated)
        ).toUpperCase()}{" "}
        AGO
      </p>
      <h3>{cityData.city}</h3>
      <p>In {cityData.location}, United Kingdom</p>
      <b>
        Values:{" "}
        {cityData.measurements
          .map(
            (measurement) =>
              measurement.parameter.toUpperCase() + ": " + measurement.value
          )
          .join(", ")}
      </b>
    </div>
  );
};

export default Card;
