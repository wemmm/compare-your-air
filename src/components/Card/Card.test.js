import { render, screen, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import Card from "./Card";

const mockCityData = {
  city: "Manchester",
  location: "Salford Eccles",
  measurements: [
    {
      lastUpdated: "2022-02-14T21:00:00Z",
      parameter: "pm10",
      value: 13,
    },
  ],
};

test("renders", () => {
  const component = renderer.create(
    <Card cityData={mockCityData} onClose={jest.fn} />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("displays data passed in as prop", () => {
  render(<Card cityData={mockCityData} onClose={jest.fn} />);

  expect(screen.getByText("Salford Eccles")).toBeInTheDocument();
  expect(screen.getByText("in Manchester, United Kingdom")).toBeInTheDocument();
  expect(screen.getByText("Values: PM10: 13")).toBeInTheDocument();
});

test("calls onClose function when close button is clicked", () => {
  const mockFunction = jest.fn();

  render(<Card cityData={mockCityData} onClose={mockFunction} />);

  fireEvent.click(screen.getByRole("button"));

  expect(mockFunction).toHaveBeenCalledTimes(1);
});
