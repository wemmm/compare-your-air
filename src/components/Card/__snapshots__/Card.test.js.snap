// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="card"
>
  <div
    className="card--close-button-container"
  >
    <button
      className="card--close-button"
      onClick={[Function]}
    >
      <svg>
        close.svg
      </svg>
    </button>
  </div>
  <p
    className="card--updated-time"
  >
    UPDATED
     
    2 DAYS
     
    AGO
  </p>
  <h3>
    Salford Eccles
  </h3>
  <p
    className="card--city"
  >
    in 
    Manchester
    , United Kingdom
  </p>
  <b>
    Values:
     
    PM10: 13
  </b>
</div>
`;
